torch/csrc/autograd/profiler.cpp

```diff
@@ -50,33 +87,35 @@ RangeEventList& getEventList() {
 }
 
 void mark(std::string name, bool include_cuda /* = true */) {
-  if (state == ProfilerState::Disabled) {
+  if (config.state == ProfilerState::Disabled) {
     return;
   }
-  if (state == ProfilerState::NVTX) {
+  if (config.state == ProfilerState::NVTX) {
     cuda_stubs->nvtxMarkA(name.c_str());
   } else {
-    getEventList().record(
+    auto& list = getEventList();
+    std::lock_guard<std::mutex> guard(all_event_lists_map_mutex);
```

Comment:
What changed such that we now require this lock here?

Reply:
I think it was generally not thread safe, e.g. you can have a situation when two threads write in the same list of events

Reply:
and, in the end, when we consolidate we still need to acq. a lock; this code changes though in the next upd when we move away from global lists of events